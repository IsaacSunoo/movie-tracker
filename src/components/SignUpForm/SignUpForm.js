import React, { Component } from 'react';

class SignUpForm extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      password: ''
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({
      [name]: value
    })
  }

  handleSignUp = async (e) => {
    e.preventDefault();
    const userInfo = this.state;
    const newUserUrl = 'http://localhost:3000/api/users/new';
    const options = {
      method: 'post',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(userInfo)
    }
    try {
      const response = await fetch(newUserUrl, options);
      const newUser = await response.json();
      console.log(newUser);
    } catch (err) {
      throw new Error(err);
    }
  }

  render() {
    const { name, email, password } = this.state;
    const { switchDisplay } = this.props;
    return (
      <div className='sign-up-form'>
          <form onSubmit={this.handleSignUp}>
            <label>Name:<input type='text' name='name' value={name} onChange={this.handleChange}/></label>
            <label>Email:<input type='email' name='email' value={email} onChange={this.handleChange}/></label>
            <label>Password:<input type='password' name='password' value={password} onChange={this.handleChange}/></label>
            <button type='submit'>Create Account</button>
          </form>
          <div onClick={() => {switchDisplay('login')}} className='pointer'>Return to Login</div>
      </div>
    )
  }
}

export default SignUpForm;