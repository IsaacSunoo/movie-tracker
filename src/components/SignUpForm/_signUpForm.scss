.sign-up-form {
    form {
        display: flex;

        input {
            border: 0;
            background-color: #a5a9d3;
            height: 25px;
            &:focus {
                outline: none;
            }
        }

        label {
            color: $cream;
            display: flex;
            flex-direction: column;
            margin-right: 20px;
        }

        button {
            @include btn-design();
            margin: 23px 20px 0 0;
        }
    }
}