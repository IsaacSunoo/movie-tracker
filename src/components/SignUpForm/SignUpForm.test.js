import React from 'react';
import SignUpForm from './SignUpForm';
import { shallow } from 'enzyme';

describe('SignUpForm', () => {
  
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <SignUpForm />
    )
  });
  
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    expect(wrapper.state()).toEqual({
      name: '',
      email: '',
      password: ''
    });
  });

  it('should handleChange', () => {

  });

  it('should handleSignUp', () => {

  });
});