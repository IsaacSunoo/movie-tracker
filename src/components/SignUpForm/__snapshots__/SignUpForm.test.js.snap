// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUpForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
        >
          <label>
            Name:
            <input
              name="name"
              onChange={[Function]}
              type="text"
              value=""
            />
          </label>
          <label>
            Email:
            <input
              name="email"
              onChange={[Function]}
              type="email"
              value=""
            />
          </label>
          <label>
            Password:
            <input
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />
          </label>
          <button
            type="submit"
          >
            Create Account
          </button>
        </form>,
        <div
          className="pointer"
          onClick={[Function]}
        >
          Return to Login
        </div>,
      ],
      "className": "sign-up-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Name:
              <input
                name="name"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>,
            <label>
              Email:
              <input
                name="email"
                onChange={[Function]}
                type="email"
                value=""
              />
            </label>,
            <label>
              Password:
              <input
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />
            </label>,
            <button
              type="submit"
            >
              Create Account
            </button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Name:",
                <input
                  name="name"
                  onChange={[Function]}
                  type="text"
                  value=""
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Name:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "name",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Email:",
                <input
                  name="email"
                  onChange={[Function]}
                  type="email"
                  value=""
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Email:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "email",
                  "onChange": [Function],
                  "type": "email",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Password:",
                <input
                  name="password"
                  onChange={[Function]}
                  type="password"
                  value=""
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Password:",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "password",
                  "onChange": [Function],
                  "type": "password",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create Account",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Create Account",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Return to Login",
          "className": "pointer",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Return to Login",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
          >
            <label>
              Name:
              <input
                name="name"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
            <label>
              Email:
              <input
                name="email"
                onChange={[Function]}
                type="email"
                value=""
              />
            </label>
            <label>
              Password:
              <input
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />
            </label>
            <button
              type="submit"
            >
              Create Account
            </button>
          </form>,
          <div
            className="pointer"
            onClick={[Function]}
          >
            Return to Login
          </div>,
        ],
        "className": "sign-up-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Name:
                <input
                  name="name"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </label>,
              <label>
                Email:
                <input
                  name="email"
                  onChange={[Function]}
                  type="email"
                  value=""
                />
              </label>,
              <label>
                Password:
                <input
                  name="password"
                  onChange={[Function]}
                  type="password"
                  value=""
                />
              </label>,
              <button
                type="submit"
              >
                Create Account
              </button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Name:",
                  <input
                    name="name"
                    onChange={[Function]}
                    type="text"
                    value=""
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Name:",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "name",
                    "onChange": [Function],
                    "type": "text",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Email:",
                  <input
                    name="email"
                    onChange={[Function]}
                    type="email"
                    value=""
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Email:",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "email",
                    "onChange": [Function],
                    "type": "email",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Password:",
                  <input
                    name="password"
                    onChange={[Function]}
                    type="password"
                    value=""
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Password:",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "password",
                    "onChange": [Function],
                    "type": "password",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Create Account",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Create Account",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Return to Login",
            "className": "pointer",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Return to Login",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
